model/modelprocessor/datastream.go

```diff
@@ -141,10 +141,18 @@ func metricsetDataset(event *modelpb.APMEvent) string {
 		// metrics that we don't already know about; otherwise they will end
 		// up creating service-specific data streams.
 		internal := true
-		for _, s := range event.Metricset.Samples {
-			if !IsInternalMetricName(s.Name) {
-				internal = false
-				break
+
+		// set internal to false for metrics translated using OTel remappers.
```

Comment:
Do we plan to allow to remap further metrics in the future? I'm worried this code could be difficult to change if we want to remap different metrics in the future. Disregard this comment if this could be easily solved just looping over `event` attributes to achieve it.

Reply:
Sorry, I didn't get this point.

> Do we plan to allow to remap further metrics in the future?

Yes we will be having more remapped metrics in the future.

> I'm worried this code could be difficult to change if we want to remap different metrics in the future. Disregard this comment if this could be easily solved just looping over event attributes to achieve it.

All remapped metrics should go to apm.app datastream so I don't see why we would require to loop over anything. Let me know if I am missing the point here.

Reply:
IIANM we'll also be remapping k8s metrics? In which case they would come from some other receiver IIANM. I wonder if we should just check for the `event.provider` key, and treat them as app metrics if the key exists regardless of the value.

Reply:
> Maybe you mean what should we do in case the remapping has a different way to identify it?

@lahsivjar yes indeed, sorry for the confusion (my comment would have been better of on the next `if` when `event.provider` is checked).

Reply:
The key for identifying remapped metrics was changed to `event.module: elastic/opentelemetry-lib` via https://github.com/elastic/opentelemetry-lib/pull/18. I have updated the code as per the changes. We will always have this key for all remapped metrics. Let me know if there are any other concerns regarding this.